notifier.rst

```diff
@@ -0,0 +1,561 @@
+                ],
+            ],
+        ]);
+
+
+Now, whenever the notification's importance is set to "high", it will be
+sent using the Slack transport::
+
+    // ...
+    class InvoiceController extends AbstractController
+    {
+        /**
+         * @Route("/invoice/create")
+         */
+        public function invoice(NotifierInterface $notifier)
+        {
+            // ...
+
+            $notification = (new Notification('New Invoice'))
+                ->content('You got a new invoice for 15 EUR.')
+                ->importance(Notification::IMPORTANCE_HIGH);
+
+            $notifier->send($notification, new Recipient('wouter@wouterj.nl'));
+
+            // ...
+        }
+    }
+
+Customize Notifications
+-----------------------
+
+You can extend the ``Notification`` or ``Recipient`` base classes to
+customize their behaviour. For instance, you can overwrite the
+``getChannels()`` method to only return ``sms`` if the invoice price is
+very high and the recipient has a phonenumber::
+
+    namespace App\Notifier;
+
+    use Symfony\Component\Notifier\Notification\Notification;
+    use Symfony\Component\Notifier\Recipient\Recipient;
+
+    class InvoiceNotification extends Notification
+    {
+        private $price;
+
+        public function __construct(int $price)
+        {
+            $this->price = $price;
+        }
+
+        public function getChannels(Recipient $recipient)
+        {
+            if ($this->price > 10000 && $receiver->hasPhone()) {
+                return ['sms', 'email'];
+            }
+
+            return 'email';
+        }
+    }
+
+Customize Notification Messages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each channel has its own notification interface that you can implement to
+customize the notification message. For instance, if you want to modify the
+message based on the chat service, implement
+:class:`Symfony\\Component\\Notifier\\Notification\\ChatNotificationInterface`
+and its ``asChatMessage()`` method::
+
+    // src/Notifier/InvoiceNotification.php
+    namespace App\Notifier;
+
+    use Symfony\Component\Notifier\Message\ChatMessage;
+    use Symfony\Component\Notifier\Notification\Notification;
+    use Symfony\Component\Notifier\Notification\ChatNotificationInterface;
+    use Symfony\Component\Notifier\Recipient\Recipient;
+
+    class InvoiceNotification extends Notification implements ChatNotificationInterface
+    {
+        private $price;
+
+        public function __construct(int $price)
+        {
+            $this->price = $price;
+        }
+
+        public function asSmsMessage(Recipient $recipient, string $transport = null): ?ChatMessage
+        {
+            // add a custom emoji if the message is sent to Slack
+            if ('slack' === $transport) {
+                return (new ChatMessage('You\'re invoiced '.$this->price.' EUR.'))
+                    ->emoji('tada');
+            }
+
+            // the default message is created if null is returned
```

Comment:
This sentence is a bit unclear to me 🤔

Reply:
I tried improving it now. The `asChatMessage()` is called to customize the ChatMessage, if it doesn't exists Symfony will create a default ChatMessage based on the notification. Returning null from `asChatMessage()` also result in this default ChatMessage being created.